saskatoon/harvest/views.py

```diff
@@ -95,31 +95,40 @@ class HarvestCreateView(LoginRequiredMixin, SuccessMessageMixin, CreateView):
     template_name = 'app/forms/model_form.html'
     success_message = _("Harvest created successfully!")
 
-    def get_context_data(self, **kwargs):
+    def get_property(self):
+        """If creating new harvest from Property page"""
+        try:
+            pid = self.request.GET['pid']
+            return Property.objects.get(id=pid)
+        except (KeyError, Property.DoesNotExist):
+            return None
+
+    def get_initial(self):
+        initial = {}
+        _property = self.get_property()
+        if _property:
+            initial['property'] = _property
+            initial['trees'] = _property.trees.all()
+
+        if 'Pick Leader' in self.request.user.roles:
+            initial['pick_leader'] = self.request.user
+        return initial
 
-        try: # registering new harvest for specific property
-            p = Property.objects.get(id=self.request.GET['pid'])
-            initial = {'property': p,
-                       'trees': p.trees.all()
-                       }
-            cancel_url = '/property/' + str(p.id)
-        except KeyError:
-            initial = {}
+    def get_context_data(self, **kwargs):
+        _property = self.get_property()
+        if _property:
+            cancel_url = reverse_lazy('property-detail',
+                                      kwargs={'pk': _property.id})
+        else:
             cancel_url = reverse_lazy('harvest-list')
 
```

Comment:
Is this the diff fixing the sporadic server errors in harvest creation?

Reply:
using `get_initial` instead of recreating the form everytime in `get_context_data` is what allows ValidationErrors to be passed to form when `is_valid` is False